From: Guido Guenther <agx@sigxcpu.org>
Date: Fri, 7 Mar 2008 09:08:26 +0100
Subject: Don't enable default network on boot

to not interfere with existing network configurations
---
 src/Makefile.am | 3 ---
 src/Makefile.in | 3 ---
 2 files changed, 6 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 87f5101..e8e5fef 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -2558,9 +2558,6 @@ if WITH_NETWORK
 	    cp $(DESTDIR)$(confdir)/qemu/networks/default.xml.t \
 	      $(DESTDIR)$(confdir)/qemu/networks/default.xml && \
 	    rm $(DESTDIR)$(confdir)/qemu/networks/default.xml.t; }
-	test -e $(DESTDIR)$(confdir)/qemu/networks/autostart/default.xml || \
-	   ln -s ../default.xml \
-	    $(DESTDIR)$(confdir)/qemu/networks/autostart/default.xml
 endif WITH_NETWORK
 
 uninstall-local:: uninstall-init uninstall-systemd
diff --git a/src/Makefile.in b/src/Makefile.in
index 166e642..3d9dd91 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -9637,9 +9637,6 @@ install-data-local: install-init install-systemd
 @WITH_NETWORK_TRUE@	    cp $(DESTDIR)$(confdir)/qemu/networks/default.xml.t \
 @WITH_NETWORK_TRUE@	      $(DESTDIR)$(confdir)/qemu/networks/default.xml && \
 @WITH_NETWORK_TRUE@	    rm $(DESTDIR)$(confdir)/qemu/networks/default.xml.t; }
-@WITH_NETWORK_TRUE@	test -e $(DESTDIR)$(confdir)/qemu/networks/autostart/default.xml || \
-@WITH_NETWORK_TRUE@	   ln -s ../default.xml \
-@WITH_NETWORK_TRUE@	    $(DESTDIR)$(confdir)/qemu/networks/autostart/default.xml
 
 uninstall-local:: uninstall-init uninstall-systemd
 @WITH_LIBVIRTD_TRUE@	rmdir "$(DESTDIR)$(localstatedir)/lib/libvirt/lockd/files" ||:
