From: Guido Guenther <agx@sigxcpu.org>
Date: Fri, 7 Mar 2008 09:08:26 +0100
Subject: Don't enable default network on boot

to not interfere with existing network configurations



---
 src/Makefile.am |    3 ---
 src/Makefile.in |    3 ---
 2 files changed, 6 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 0dadc29..998fd78 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1625,9 +1625,6 @@ if WITH_NETWORK
 	    cp $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml.t \
 	      $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml && \
 	    rm $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml.t; }
-	test -e $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/autostart/default.xml || \
-	   ln -s ../default.xml \
-	    $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/autostart/default.xml
 endif
 
 uninstall-local::
diff --git a/src/Makefile.in b/src/Makefile.in
index 209e118..90598de 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -7486,9 +7486,6 @@ install-data-local:
 @WITH_NETWORK_TRUE@	    cp $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml.t \
 @WITH_NETWORK_TRUE@	      $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml && \
 @WITH_NETWORK_TRUE@	    rm $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/default.xml.t; }
-@WITH_NETWORK_TRUE@	test -e $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/autostart/default.xml || \
-@WITH_NETWORK_TRUE@	   ln -s ../default.xml \
-@WITH_NETWORK_TRUE@	    $(DESTDIR)$(sysconfdir)/libvirt/qemu/networks/autostart/default.xml
 
 uninstall-local::
 	rmdir "$(DESTDIR)$(localstatedir)/cache/libvirt" ||:
