From 25ec2f6edb92ec663ebd567f12b94f3b903ebce0 Mon Sep 17 00:00:00 2001
From: Benedikt Spranger <b.spranger@linutronix.de>
Date: Thu, 23 Aug 2018 15:56:49 +0200
Subject: [PATCH 1/9] fix_driver_impl_check

---
 src/check-driverimpls.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/check-driverimpls.pl b/src/check-driverimpls.pl
index 3c0d54724c..7010dcc820 100755
--- a/src/check-driverimpls.pl
+++ b/src/check-driverimpls.pl
@@ -41,7 +41,7 @@ while (<>) {
 
             my $suffix = $impl;
             my $prefix = $impl;
-            $prefix =~ s/^([a-z]+)(.*?)$/$1/;
+            $prefix =~ s/^([a-z][a-z0-9]*)(.*?)$/$1/;
 
             if (defined $mainprefix) {
                 if ($mainprefix ne $prefix) {
@@ -53,7 +53,7 @@ while (<>) {
             }
 
             if ($api !~ /^$mainprefix/) {
-                $suffix =~ s/^[a-z]+//;
+                $suffix =~ s/^[a-z][a-z0-9]*//;
                 $suffix =~ s/^([A-Z]+)/lc $1/e;
             }
 
-- 
2.23.0

