From 58cd2086416453f98991c2d7346b40a70f9e295a Mon Sep 17 00:00:00 2001
From: Benedikt Spranger <b.spranger@linutronix.de>
Date: Wed, 22 May 2019 22:17:48 +0200
Subject: [PATCH 01/10] fix_driver_impl_check

---
 src/check-driverimpls.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/check-driverimpls.pl b/src/check-driverimpls.pl
index 832a3900f4..87887ba3f8 100755
--- a/src/check-driverimpls.pl
+++ b/src/check-driverimpls.pl
@@ -40,7 +40,7 @@ while (<>) {
 
             my $suffix = $impl;
             my $prefix = $impl;
-            $prefix =~ s/^([a-z]+(?:Unified)?)(.*?)$/$1/;
+            $prefix =~ s/^([a-z][a-z0-9]*(?:Unified)?)(.*?)$/$1/;
 
             if (defined $mainprefix) {
                 if ($mainprefix ne $prefix) {
@@ -52,7 +52,7 @@ while (<>) {
             }
 
             if ($api !~ /^$mainprefix/) {
-                $suffix =~ s/^[a-z]+(?:Unified)?//;
+                $suffix =~ s/^[a-z][a-z0-9]*(?:Unified)?//;
                 $suffix =~ s/^([A-Z]+)/lc $1/e;
             }
 
-- 
2.19.0

