From 89bc376242111d3e215a8134e7de14a273219096 Mon Sep 17 00:00:00 2001
From: Benedikt Spranger <b.spranger@linutronix.de>
Date: Thu, 23 Aug 2018 15:56:50 +0200
Subject: [PATCH 7/9] r4d_gracefull_shutdown

---
 src/r4d/r4d_driver.c | 59 +++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 58 insertions(+), 1 deletion(-)

diff --git a/src/r4d/r4d_driver.c b/src/r4d/r4d_driver.c
index 291c0ef8a4..d705637804 100644
--- a/src/r4d/r4d_driver.c
+++ b/src/r4d/r4d_driver.c
@@ -446,13 +446,68 @@ r4dDomainDestroy(virDomainPtr dom)
     return r4dDomainDestroyFlags(dom, 0);
 }
 
+/* domainShutdownFlags */
+static int r4dDomainShutdownFlags(virDomainPtr dom,
+				  unsigned int flags ATTRIBUTE_UNUSED)
+{
+    struct r4d_driver *driver = dom->conn->privateData;
+    int ret = -1, status;
+
+    logger (__func__);
+    virCheckFlags (0, -1);
+
+    status = r4d_power_status (driver->endpoint, dom->name);
+    if (!status) {
+        virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                       _("domain is not in powerup state"));
+        goto cleanup;
+    }
+
+    ret = r4d_shutdown (driver->endpoint, dom->name);
+    if (ret)
+        virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                       _("fail to shutdown domain"));
+
+cleanup:
+    return ret;
+}
+
+/* domainShutdown */
+static int r4dDomainShutdown(virDomainPtr dom)
+{
+	return r4dDomainShutdownFlags(dom, 0);
+}
+
 /* domainReboot */
 static int
 r4dDomainReboot(virDomainPtr dom ATTRIBUTE_UNUSED,
                 unsigned int flags ATTRIBUTE_UNUSED)
 {
+    struct r4d_driver *driver = dom->conn->privateData;
+    int ret = -1, status;
+
     logger (__func__);
-    return 0;
+    virCheckFlags (0, -1);
+
+    status = r4d_power_status (driver->endpoint, dom->name);
+    if (!status) {
+        virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                       _("domain is not in powerup state"));
+        goto cleanup;
+    }
+
+    ret = r4d_shutdown (driver->endpoint, dom->name);
+    if (ret)
+        virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                       _("fail to shutdown domain"));
+
+    ret = r4d_power_change (driver->endpoint, dom->name, 1);
+    if (ret)
+        virReportError(VIR_ERR_INTERNAL_ERROR, "%s",
+                       _("fail to powerup domain"));
+
+cleanup:
+    return ret;
 }
 
 /* domainReset */
@@ -677,6 +732,8 @@ static virHypervisorDriver r4dHypervisorDriver = {
     .domainCreateWithFlags =  r4dDomainCreateWithFlags, /* 1.3.1 */
     .domainDestroy =  r4dDomainDestroy, /* 1.3.1 */
     .domainDestroyFlags =  r4dDomainDestroyFlags, /* 1.3.1 */
+    .domainShutdown = r4dDomainShutdown, /* 1.3.1 */
+    .domainShutdownFlags = r4dDomainShutdownFlags, /* 1.3.1 */
     .domainReboot =  r4dDomainReboot, /* 1.3.1 */
     .domainReset =  r4dDomainReset, /* 1.3.1 */
 
-- 
2.23.0

