From: Guido Guenther <agx@sigxcpu.org>
Date: Fri, 7 Mar 2008 09:08:25 +0100
Subject: [PATCH] remove-RHism.diff

---
 tools/virsh.pod |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/tools/virsh.pod b/tools/virsh.pod
index 6ff0151..7fafe13 100644
--- a/tools/virsh.pod
+++ b/tools/virsh.pod
@@ -34,7 +34,7 @@ the program.
 
 All B<virsh> operations rely upon the libvirt library.
 For any virsh commands to run xend/qemu, or what ever virtual library that libvirt supports.  For this reason you should start xend/qemu as a service when your system first boots using xen/qemu. This can usually be done using the command
-B<service libvirtd start> .
+B<invoke-rc.d libvirt-bin start> .
 
 Most B<virsh> commands require root privileges to run due to the
 communications channels used to talk to the hypervisor.  Running as
-- 
