From 0b5d455fb497a1d3785d862639102697eda4d2e3 Mon Sep 17 00:00:00 2001
From: Cole Robinson <crobinso@redhat.com>
Date: Mon, 22 Sep 2008 16:03:00 +0200
Subject: [PATCH] fix crash when no emulator is defined for kvm/qemu

---
 src/qemu_conf.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/qemu_conf.c b/src/qemu_conf.c
index d742c32..23ef050 100644
--- a/src/qemu_conf.c
+++ b/src/qemu_conf.c
@@ -790,7 +790,7 @@ int qemudBuildCommandLine(virConnectPtr conn,
     if (!emulator)
         return -1;
 
-    ADD_ARG_LIT(vm->def->emulator);
+    ADD_ARG_LIT(emulator);
     ADD_ARG_LIT("-S");
     ADD_ARG_LIT("-M");
     ADD_ARG_LIT(vm->def->os.machine);
-- 
1.5.6.5

