From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 11 Feb 2010 19:02:32 +0100
Subject: [PATCH] Don't drop caps when exec'ing qemu

Closes: #565767
---
 src/qemu/qemu_driver.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index df1d435..ce1c3cf 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -3363,7 +3363,7 @@ static int qemudStartVMDaemon(virConnectPtr conn,
 
     ret = virExecDaemonize(argv, progenv, &keepfd, &child,
                            stdin_fd, &logfile, &logfile,
-                           VIR_EXEC_NONBLOCK | VIR_EXEC_CLEAR_CAPS,
+                           VIR_EXEC_NONBLOCK /* | VIR_EXEC_CLEAR_CAPS */,
                            qemudSecurityHook, &hookData,
                            pidfile);
     VIR_FREE(pidfile);
-- 
