From: Gabor Gombas <gombasg@sztaki.hu>
Date: Fri, 8 Jan 2010 11:00:07 +0100
Subject: [PATCH] Terminate nc on EOF

Closes: #564053
---
 src/remote/remote_driver.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/remote/remote_driver.c b/src/remote/remote_driver.c
index 317125f..2bb5a99 100644
--- a/src/remote/remote_driver.c
+++ b/src/remote/remote_driver.c
@@ -736,7 +736,7 @@ doRemoteOpen (virConnectPtr conn,
     }
 
     case trans_ssh: {
-        int j, nr_args = 6;
+        int j, nr_args = 8;
 
         if (username) nr_args += 2; /* For -l username */
         if (no_tty) nr_args += 5;   /* For -T -o BatchMode=yes -e none */
@@ -770,6 +770,8 @@ doRemoteOpen (virConnectPtr conn,
         }
         cmd_argv[j++] = strdup (priv->hostname);
         cmd_argv[j++] = strdup (netcat ? netcat : "nc");
+        cmd_argv[j++] = strdup ("-q");
+        cmd_argv[j++] = strdup ("0");
         cmd_argv[j++] = strdup ("-U");
         cmd_argv[j++] = strdup (sockname ? sockname :
                                 (flags & VIR_CONNECT_RO
-- 
