From de516e669fcc0e4d1268698dd0a48e9c79da1eaf Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 4 Dec 2008 11:51:13 +0100
Subject: [PATCH] also look for /usr/bin/kvm

---
 src/qemu_driver.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/qemu_driver.c b/src/qemu_driver.c
index 58530aa..0d5e768 100644
--- a/src/qemu_driver.c
+++ b/src/qemu_driver.c
@@ -1749,6 +1749,7 @@ static const char *qemudProbe(void)
 {
     if ((virFileExists("/usr/bin/qemu")) ||
         (virFileExists("/usr/bin/qemu-kvm")) ||
+        (virFileExists("/usr/bin/kvm")) ||
         (virFileExists("/usr/bin/xenner"))) {
         if (getuid() == 0) {
             return("qemu:///system");
-- 
1.6.0.3

