From: =?utf-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 23 Jan 2009 16:25:33 +0100
Subject: [PATCH] don't fail on missing locales

Closes: #512721
---
 src/virsh.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/virsh.c b/src/virsh.c
index 2b05793..5e39b48 100644
--- a/src/virsh.c
+++ b/src/virsh.c
@@ -7078,7 +7078,7 @@ main(int argc, char **argv)
 
     if (!setlocale(LC_ALL, "")) {
         perror("setlocale");
-        return -1;
+        /* failure to setup locale is not fatal */
     }
     if (!bindtextdomain(GETTEXT_PACKAGE, LOCALEBASEDIR)) {
         perror("bindtextdomain");
-- 
