From b797c0fc100d75b8252b594670d007db888a10bd Mon Sep 17 00:00:00 2001
From: Cole Robinson <crobinso@redhat.com>
Date: Mon, 1 Dec 2008 17:36:51 -0500
Subject: [PATCH] Increase initial qemu monitor read timeout

Closes: #499720
---
 src/qemu_driver.c |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/qemu_driver.c b/src/qemu_driver.c
index 0d5e768..ffe2dd2 100644
--- a/src/qemu_driver.c
+++ b/src/qemu_driver.c
@@ -418,9 +418,9 @@ qemudReadMonitorOutput(virConnectPtr conn,
                        char *buf,
                        int buflen,
                        qemudHandlerMonitorOutput func,
-                       const char *what)
+                       const char *what,
+                       int timeout)
 {
-#define MONITOR_TIMEOUT 3000
     int got = 0;
     buf[0] = '\0';
 
@@ -446,7 +446,7 @@ qemudReadMonitorOutput(virConnectPtr conn,
                 return -1;
             }
 
-            ret = poll(&pfd, 1, MONITOR_TIMEOUT);
+            ret = poll(&pfd, 1, timeout);
             if (ret == 0) {
                 qemudReportError(conn, NULL, NULL, VIR_ERR_INTERNAL_ERROR,
                                  _("Timed out while reading %s startup output"), what);
@@ -480,7 +480,6 @@ qemudReadMonitorOutput(virConnectPtr conn,
                      _("Out of space while reading %s startup output"), what);
     return -1;
 
-#undef MONITOR_TIMEOUT
 }
 
 static int
@@ -526,7 +525,7 @@ static int qemudOpenMonitor(virConnectPtr conn,
                                  driver, vm, monfd,
                                  buf, sizeof(buf),
                                  qemudCheckMonitorPrompt,
-                                 "monitor");
+                                 "monitor", 10000);
 
     /* Keep monitor open upon success */
     if (ret == 0)
@@ -640,7 +639,7 @@ static int qemudWaitForMonitor(virConnectPtr conn,
                                      driver, vm, vm->stderr_fd,
                                      buf, sizeof(buf),
                                      qemudFindCharDevicePTYs,
-                                     "console");
+                                     "console", 3000);
 
     buf[sizeof(buf)-1] = '\0';
 
-- 
1.6.0.3

