From ed71be2693133d640b4a99eb4b6f90c51bdffff8 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 4 Dec 2008 16:22:52 +0100
Subject: [PATCH] raise error on invalid volume format

---
 src/storage_conf.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/storage_conf.c b/src/storage_conf.c
index 29d74c8..f0d2b94 100644
--- a/src/storage_conf.c
+++ b/src/storage_conf.c
@@ -753,6 +753,8 @@ virStorageVolDefParseDoc(virConnectPtr conn,
     if (options->formatFromString) {
         char *format = virXPathString(conn, "string(/volume/target/format/@type)", ctxt);
         if ((ret->target.format = (options->formatFromString)(conn, format)) < 0) {
+            virStorageReportError(conn, VIR_ERR_XML_ERROR,
+                                  _("unknown volume format type %s"), format);
             VIR_FREE(format);
             goto cleanup;
         }
-- 
1.6.0.3

