From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 26 Feb 2009 14:29:58 +0100
Subject: [PATCH] fix Debian specific path to hvm loader

Closes: #517059
---
 src/xen_internal.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/xen_internal.c b/src/xen_internal.c
index d80ecfb..d04fb4f 100644
--- a/src/xen_internal.c
+++ b/src/xen_internal.c
@@ -2204,7 +2204,7 @@ xenHypervisorBuildCapabilities(virConnectPtr conn,
                                               "/usr/lib64/xen/bin/qemu-dm" :
                                               "/usr/lib/xen/bin/qemu-dm"),
                                              (guest_archs[i].hvm ?
-                                              "/usr/lib/xen/boot/hvmloader" :
+                                              "/usr/lib/xen-default/boot/hvmloader" :
                                               NULL),
                                              1,
                                              machines)) == NULL)
-- 
