From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Tue, 17 Nov 2015 17:05:30 +0100
Subject: Disable service timeout for libvirt-guests

Closes: #803714
---
 tools/libvirt-guests.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/libvirt-guests.service.in b/tools/libvirt-guests.service.in
index fc353f0..cb1d7bd 100644
--- a/tools/libvirt-guests.service.in
+++ b/tools/libvirt-guests.service.in
@@ -13,6 +13,7 @@ ExecStop=@libexecdir@/libvirt-guests.sh stop
 Type=oneshot
 RemainAfterExit=yes
 StandardOutput=journal+console
+TimeoutStopSec=0
 
 [Install]
 WantedBy=multi-user.target
