From fefae8c13aa89d3e9fe0843a3af9fef574b67127 Mon Sep 17 00:00:00 2001
From: Guido Guenther <agx@sigxcpu.org>
Date: Fri, 7 Mar 2008 09:08:27 +0100
Subject: [PATCH] libvirt-group.diff

---
 qemud/libvirtd.conf |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/qemud/libvirtd.conf b/qemud/libvirtd.conf
index 3b7db4d..2b15365 100644
--- a/qemud/libvirtd.conf
+++ b/qemud/libvirtd.conf
@@ -70,7 +70,7 @@
 # without becoming root.
 #
 # This is restricted to 'root' by default.
-#unix_sock_group = "libvirt"
+unix_sock_group = "libvirt"
 
 # Set the UNIX socket permissions for the R/O socket. This is used
 # for monitoring VM status only
@@ -87,7 +87,7 @@
 #
 # If not using PolicyKit and setting group ownership for access
 # control then you may want to relax this to:
-#unix_sock_rw_perms = "0770"
+unix_sock_rw_perms = "0770"
 
 
 
-- 
1.5.4.3

