From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Mon, 7 Apr 2014 11:11:07 +0200
Subject: libvirt-guests: avoid bashism

---
 tools/libvirt-guests.sh.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/libvirt-guests.sh.in b/tools/libvirt-guests.sh.in
index 116b232..62ee35f 100644
--- a/tools/libvirt-guests.sh.in
+++ b/tools/libvirt-guests.sh.in
@@ -101,13 +101,15 @@ test_connect()
 {
     uri=$1
 
-    for ((i = 0; i < ${CONNECT_RETRIES}; i++)); do
+    i=${CONNECT_RETRIES}
+    while [ $i -gt 0 ]; do
         run_virsh "$uri" connect 2>/dev/null
         if [ $? -eq 0 ]; then
             return 0;
         fi
         sleep ${RETRIES_SLEEP}
         eval_gettext "Unable to connect to libvirt currently. Retrying .. \$i"
+        i=$(($i-1))
     done
     eval_gettext "Can't connect to \$uri. Skipping."
     echo
