diff --git a/src/qemu_conf.c b/src/qemu_conf.c
index 48ccc60..bed5bf7 100644
--- a/src/qemu_conf.c
+++ b/src/qemu_conf.c
@@ -2524,7 +2525,7 @@ qemudLoadConfig(struct qemud_driver *driver,
     if (!(def = qemudParseVMDef(NULL, driver, xml, file))) {
         virErrorPtr err = virGetLastError();
         qemudLog(QEMUD_WARN, "Error parsing QEMU guest config '%s' : %s",
-                 path, err->message);
+                 path, err ? err->message : "unknown error\n" );
         return NULL;
     }
 
